app/js/HttpController.js

```diff
@@ -188,6 +188,27 @@ module.exports = HttpController = {
     })
   },
 
+  patchDoc(req, res, next) {
+    const { project_id, doc_id } = req.params
+    logger.log({ project_id, doc_id }, 'patching doc')
+
+    const allowedFields = ['deleted', 'deletedAt', 'name']
+    const meta = {}
+    Object.entries(req.body).forEach(([field, value]) => {
+      if (allowedFields.includes(field)) {
+        meta[field] = value
+      } else {
+        logger.fatal({ field }, 'joi validation for pathDoc is broken')
+      }
+    })
```

Comment:
oh, ok. I see why you feel it's redoing validation. What I had in mind was something like:
```js
const { deleted, deletedAt, name } = req.body
DocManager.patchDoc(project_id, doc_id, { deleted, deletedAt, name }, ...)
```
but let's not quibble further over that. Do what you feel is best: this version, the previous version, or what I suggest above.

Reply:
The simple object destruction and re-composing would possibly insert `undefined` values for fields that are not set. The lodash snippet is nice, but then we will never know that joi is misbehaving.

I prefer removing that commit, but I'll go with the current version to get this going possibly today.